packages/backend-tasks/src/tasks/types.ts

```diff
@@ -91,6 +75,43 @@ export interface TaskDefinition {
   initialDelay?: Duration;
 }
 
+/**
+ * Options that apply to the invocation of a given task.
+ *
+ * @public
+ */
+export interface TaskInvocationDefinition {
+  /**
+   * A unique ID (within the scope of the plugin) for the task.
+   */
+  id: string;
+
+  /**
+   * The actual task function to be invoked regularly.
+   */
+  fn: TaskFunction;
+
+  /**
+   * An abort signal that, when triggered, will stop the recurring execution of
+   * the task.
+   */
+  signal?: AbortSignal;
+}
+
+/**
+ * A previously prepared task schedule, ready to be invoked.
+ *
+ * @public
+ */
+export interface TaskSchedule {
```

Comment:
Feeling the naming here could be improved, looking at the interface it doesn't quite match up. Potentially something like `TaskRunner` & `createScheduledTaskRunner`?

Reply:
Sure, that makes sense!